app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use App\Models\Authentication;
+use App\Http\Controllers\Controller;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Foundation\Auth\RegistersUsers;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Support\Facades\DB;
+
+
+
+class RegisterController extends Controller
+{
+    /*
+    |--------------------------------------------------------------------------
+    | Register Controller
+    |--------------------------------------------------------------------------
+    |
+    | This controller handles the registration of new users as well as their
+    | validation and creation. By default this controller uses a trait to
+    | provide this functionality without requiring any additional code.
+    |
+    */
+
+    public function create() {
+        return view('pre_register');
+    }
+
+    public function completeEmail(){
+        return view('finish_pre_register');
+    }
+
+    protected function email_validator(array $data)
+    {
+        return Validator::make($data, [
+            'email' => 'required|string|email||unique:users,email',
```

Comment:
```suggestion
            'email' => 'required|string|email|unique:users,email',
```
不要なバーティカルバーが挿入されています。削除お願いします。

Reply:
この行数であれば関数として書き出すメリットがあまりありません。
メール送信しているアクションでvaldationを行うようにしてください。

Reply:
メール送信をおこなっている箇所にvalidationを移動しました。